File: BFS/c#/Bfs.cs
using System.Collections.Generic;

public static class Bfs
{
    public static List<int> Run(Graph graph, int startVertex)
    {
        if (!graph.Vertices.ContainsKey(startVertex))
            return new List<int>();

        var visited = new HashSet<int>();
        var result = new List<int>();
        var queue = new Queue<int>();
        
        queue.Enqueue(startVertex);

        while (queue.Count > 0)
        {
            int current = queue.Dequeue();
            
            if (!visited.Contains(current))
            {
                visited.Add(current);
                result.Add(current);

                if (graph.Vertices.TryGetValue(current, out var neighbors))
                {
                    foreach (int neighbor in neighbors)
                    {
                        if (!visited.Contains(neighbor))
                            queue.Enqueue(neighbor);
                    }
                }
            }
        }

        return result;
    }
}

File: BFS/java/Bfs.java
import java.util.*;

public class Bfs {
    public static List<Integer> run(Graph graph, int startVertex) {
        if (!graph.getVertices().containsKey(startVertex)) {
            return new ArrayList<>();
        }

        Set<Integer> visited = new HashSet<>();
        List<Integer> result = new ArrayList<>();
        Queue<Integer> queue = new LinkedList<>();

        queue.add(startVertex);

        while (!queue.isEmpty()) {
            int current = queue.poll();

            if (!visited.contains(current)) {
                visited.add(current);
                result.add(current);

                List<Integer> neighbors = graph.getVertices().get(current);
                if (neighbors != null) {
                    for (int neighbor : neighbors) {
                        if (!visited.contains(neighbor)) {
                            queue.add(neighbor);
                        }
                    }
                }
            }
        }

        return result;
    }
}
